cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -422,18 +444,11 @@ public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile,
    *                               the artifact is added successfully, then the user gets {@link Action#ALL} privileges
    *                               on the added artifact.
    */
+  @VisibleForTesting
   public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile,
```

Comment:
I actually thought about that. Do we need to expose this in TestBase? That's the only place outside of this package where this (and a 1 or two other) methods are used. I don't think this is something we could fix right now though.